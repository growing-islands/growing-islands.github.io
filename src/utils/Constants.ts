type BreakpointsType = {
  PC: number
  SP: number
}

export const BREAKPOINTS: BreakpointsType = {
  PC: 769,
  SP: 768,
}

export const PAGE_INFO = {
  URL: process.env.NEXT_PUBLIC_SITE_URL || 'https://',
  TWITTER: 'https://twitter.com/yumelab1214',
  FACEBOOK: 'https://www.facebook.com/yumelab.co.jp/',
}

export const INTERSECTION_OPTIONS = {
  root: null,
  rootMargin: '0px 0px -100%',
  threshold: 0,
}
