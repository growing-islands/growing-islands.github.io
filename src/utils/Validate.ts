/**
 * Form Validate関連の変数
 */

export const ERROR_MESSAGES = {
  required: 'この項目は入力必須です',
  maxLength: '入力できる文字数を超えています',
  mixLength: '文字数が足りません',
  max: '数値が大きすぎます',
  mix: '数値が小さすぎます',
  num: '半角数字で入力して下さい',
  hiragana: 'ひらがなで入力してください',
  fullKana: '全角カタカナで入力してください',
  tel: '電話番号が正しい形式ではありません',
  email: 'メールアドレスの形式が正しくありません',
  password:
    '英文字（大文字、小文字）、数字をそれぞれ最低一文字以上入れた8〜16文字で入力してください',
  dependent_char: '使用できない文字列が含まれています',
  url: '「http://」もしくは「https://」から始まるURLを入力してください',
}

// NOTE： 正規表現を記述するためにESLintを無効
/*eslint-disable no-useless-escape */
/*eslint-disable no-irregular-whitespace */
export const REGEX = {
  num: /^[0-9]+$/,
  alphanumeric: /^[0-9a-zA-Z]+$/,
  hiragana: /^[ぁ-んー　]*$/,
  fullKana: /^[ァ-ヶー　]*$/,
  halfKana: /^[ｦ-ﾟ]*$/,
  tel: /^(0{1}\d{9,10})$/,
  email:
    /^([\w!#$%&'*+\-\/=?^`{|}~]+(\.[\w!#$%&'*+\-\/=?^`{|}~]+)*|"([\w!#$%&'*+\-\/=?^`{|}~. ()<>\[\]:;@,]|\\[\\"])+")@(([a-zA-Z\d\-]+\.)+[a-zA-Z]+|\[(\d{1,3}(\.\d{1,3}){3}|IPv6:[\da-fA-F]{0,4}(:[\da-fA-F]{0,4}){1,5}(:\d{1,3}(\.\d{1,3}){3}|(:[\da-fA-F]{0,4}){0,2}))\])$/,
  password:
    /^(?=.*?[a-z])(?=.*?[A-Z])(?=.*?\d)[a-zA-Z0-9!"#$%&'()\*\+\-\.,\/:;<=>?@\[\\\]^_`{|}~]{8,16}$/,
  dependent_char:
    /^(?!.*(\u2460|\u2461|\u2462|\u2463|\u2464|\u2465|\u2466|\u2467|\u2468|\u2469|\u246a|\u246b|\u246c|\u246d|\u246f|\u2470|\u2471|\u2472|\u2473|\u2160|\u2161|\u2162|\u2163|\u2164|\u2165|\u2166|\u2167|\u2168|\u2169|\u3349|\u3314|\u3322|\u334d|\u3318|\u3327|\u3303|\u3336|\u3351|\u3357|\u330d|\u3326|\u3323|\u332b|\u334a|\u333b|\u339c|\u339d|\u339e|\u338e|\u338f|\u33c4|\u33a1|\u337b|\u301d|\u301f|\u2116|\u33cd|\u2121|\u32a4|\u32a5|\u32a6|\u32a7|\u32a8|\u3231|\u3232|\u3239|\u337e|\u337d|\u337c|\u222e|\u221f|\u22bf|\u2170|\u2171|\u2172|\u2173|\u2174|\u2175|\u2176|\u2177|\u2178|\u2179|\uffe2|\uffe4|[\uD800-\uDBFF][\uDC00-\uDFFF])).*$/,
  url: /^(https?)(:\/\/[-_.!~*\'()a-zA-Z0-9;\/?:\@&=+\$,%#]+)$/,
}
/*eslint-enable no-useless-escape */
/*eslint-enable no-irregular-whitespace */
