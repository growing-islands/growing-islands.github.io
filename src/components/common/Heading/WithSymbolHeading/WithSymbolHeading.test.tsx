import { render } from '~tests/libs/utils'
import { WithSymbolHeading } from '.'

describe('components/common/Heading/WithSymbolHeading', () => {
  it('Snapshot', () => {
    const { asFragment } = render(<WithSymbolHeading text={'test'} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
