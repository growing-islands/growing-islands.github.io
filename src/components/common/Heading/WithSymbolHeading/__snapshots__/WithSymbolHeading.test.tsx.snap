// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/Heading/WithSymbolHeading Snapshot 1`] = `
<DocumentFragment>
  <header
    class=""
  >
    <svg
      aria-label="株式会社グローイングアイランズのロゴシンボル"
      class=""
      viewBox="0 0 15.63 17.45"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M15.63,2a2,2,0,1,1-2-2A2,2,0,0,1,15.63,2Z"
        />
        <path
          d="M11.54,7l.32.27a1.73,1.73,0,0,0,.48-1.21,1.78,1.78,0,0,0-1.78-1.78A1.8,1.8,0,0,0,9,5.22l.2.14Z"
        />
        <path
          d="M11.86,7.3a1.75,1.75,0,0,1-1.3.57A1.78,1.78,0,0,1,8.78,6.09,1.81,1.81,0,0,1,9,5.22L6.87,3.69A2.62,2.62,0,0,0,4.1,3.42a2.65,2.65,0,0,0-.93.8,3.2,3.2,0,0,0-.55,1.84s0,3.59,0,4.53L.88,13A1,1,0,0,0,0,14a1.11,1.11,0,0,0,.48.9L3,16.71a1.07,1.07,0,0,0,1.5-.19,1,1,0,0,0,.15-.78l1.79-2.5,4.21-1.42h0a3.17,3.17,0,0,0,1.57-1.12,2.55,2.55,0,0,0-.36-3.4Zm-1.43,3.87L6,12.66l-2.15,3a.38.38,0,0,1,.08.46.38.38,0,0,1-.53,0L.88,14.34a.38.38,0,0,1-.14-.51.37.37,0,0,1,.46-.07l2.12-2.95V9.94H5c.43,0,.63-.4.22-.79a1.42,1.42,0,0,1,.94-2.49A1.43,1.43,0,0,1,7.54,8.08a1.41,1.41,0,0,1-.48,1.07c-.42.39-.21.79.22.79h4.58A2.32,2.32,0,0,1,10.43,11.17Z"
        />
      </g>
      <ellipse
        cx="7.1"
        cy="16.93"
        rx="4.43"
        ry="0.52"
      />
    </svg>
    <h3>
      test
    </h3>
  </header>
</DocumentFragment>
`;
